frame/curve-amm/src/lib.rs

```diff
@@ -124,6 +124,17 @@ pub mod pallet {
 			+ Mutate<Self::AccountId, Balance = Self::Balance, AssetId = <Self as Config>::AssetId>
 			+ Inspect<Self::AccountId, Balance = Self::Balance, AssetId = <Self as Config>::AssetId>;
 		type Precision: Get<FixedU128>;
+		type PoolId: FullCodec
```

Comment:
I would prefer it is encoded like this 

pub trait AssetIdLike: FullCodec + Copy + Eq + PartialEq + Debug + TypeInfo {}
impl<T: FullCodec + Copy + Eq + PartialEq + Debug + TypeInfo> AssetIdLike for T {}

Reply:
What is benefit? Also if we follow that convention we may want to change every other pallets too.